internal/app/machined/internal/phase/phase.go

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package phase
+
+import (
+	"log"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/talos-systems/talos/internal/app/machined/internal/platform"
+	"github.com/talos-systems/talos/internal/app/machined/internal/platform/container"
+	"github.com/talos-systems/talos/internal/app/machined/internal/runtime"
+	"github.com/talos-systems/talos/internal/pkg/kmsg"
+	"github.com/talos-systems/talos/pkg/userdata"
+)
+
+// Runner represents a management layer for phases.
+type Runner struct {
+	platform platform.Platform
+	phases   []*Phase
+	mode     runtime.Mode
+	data     *userdata.UserData
+}
+
+// NewRunner initializes and returns a Runner.
+func NewRunner(data *userdata.UserData) (*Runner, error) {
+	platform, err := platform.NewPlatform()
+	if err != nil {
+		return nil, err
+	}
+
+	mode := runtime.Standard
+	switch platform.(type) {
+	case *container.Container:
+		mode = runtime.Container
+	default:
```

Comment:
nit, seems kind of odd to overload default to do kmsg.Setup instead of being focused on mode/platform.
unfortunately the alternative would feel awkward as well i think, so perhaps this is the better.

Reply:
This was mostly because I don't know what `log`'s default are and rather than mucking with those I chose to setup logging to `/dev/kmsg` only when we need to.